### PR TITLE
Correct set_train_test_split and first test on Silvretta 

### PR DESCRIPTION
**Pull request by marvande**
**Correction**: 
Before, in `set_train_test_split,` a random split was done for training and testing data, but this also split months of a stake belonging to the same measurement ID into test and training. We want the measurement IDs to belong fully to a fold; otherwise, the loss function doesn't work. Now it uses GroupShuffleSplit from sklearn and also launches an error if a measurement ID is split amongst folds. 

**Aggregated Predictions**: added a function `aggrPredict` to the xgb regressor that allows to give prediction in the format of measurements (i.e., annual, winter or whatever is used as ID).

**First test on Silvretta**: Code for CH region, first tests using XGB on Silvretta, correction of pre-processing of winter MB data and pulling from OGGM.

### CODE REVIEW

File: massbalancemachine/data_processing/transform_to_monthly.py [hunk 1]
```diff
@@ -97,6 +97,7 @@ def _get_column_names(vois_topographical: "list[str]") -> "list[str]":
         "POINT_ID",
         "YEAR",
         "N_MONTHS",
+        "PERIOD",
```
Comment (JulianBiesheuvel) [inline massbalancemachine/data_processing/transform_to_monthly.py:100]: Are you sure this is not just a column name specifically for CH data? Otherwise, this might not work for other data (that is supposed to be in the WGMS format).

**Review by JulianBiesheuvel — CHANGES_REQUESTED** [review]
Very nice work! :)

**Comment by marvande** [comment]
@JulianBiesheuvel, I've made an edit so that the metadata columns are optional in case someone has more metadata than the WGMS format. Now, your ML notebooks work by default (there is no need to specify the columns), and mine works with specified columns as well. Can you have a look and see if it's ok now?

**Review by JulianBiesheuvel — APPROVED** [review]
Looks good!
